FordLabs/retroquest
# Add board owners settings

## What was done
* Updated the settings modal to show an "Add Board Owners" email form when there are no emails associated with a team. If there are emails added already, it will show them instead of the form.
* Submitting the form should save one or both emails to the team in question, then show the list of emails added and hide the form.
* A cypress test was added to test this flow.
* And endpoint was added just for e2e tests that creates a team without emails. This endpoint was used in said cypress tests.

### Demo
<img width="493" alt="Screen Shot 2022-10-03 at 11 24 00 AM" src="https://user-images.githubusercontent.com/17144844/193616279-b00e5edf-d0ab-425b-9ff8-2b581da77f98.png">

<img width="473" alt="Screen Shot 2022-10-03 at 2 41 08 PM" src="https://user-images.githubusercontent.com/74197928/193653822-cd36c928-e33d-4088-9c6b-4b1bb05eda0c.png">


## Testing Instructions
This is hard to test locally. You'll have to comment out the conditionals in `AccountTab.tsx` in order to see the form since you can't create a team without an email anymore through the app.

<img width="617" alt="Screen Shot 2022-10-03 at 11 35 44 AM" src="https://user-images.githubusercontent.com/17144844/193618228-be46d91a-9dcd-4674-a4b7-9dd658649452.png">

But,

1) Ensure that you do not see the form, if the team has an email, but that you see the emails associated with the team
2) Once you remove the conditionals, check out the add board owners form. Add emails, save them. Ensure they save and that a websocket event is triggered which updates the team in the global state and thus the ui.

The flow *should* be that if the team has no email like some teams in dev, it will see the form, and once an email is saved, it will see just the emails.
